### ISSUE TITLE
Systemd is constantly unmounting my /boot partition

### ISSUE THREAD

**Issue by roberdorfer** [systemd/systemd#1378]
My sda1 is mounted to /boot after booting, and I can manually mount it just fine. But systemd is automatically unmounting that partition, even after I've manually mounted it.

These are the current permissions when it's unmounted:

```
drwxr-xr-x   2 root root    0 Sep 24 14:15 boot/
```

and when it's mounted:

```
drwxr-x---   4 root root 2.0K Jan  1  1970 boot/
```

This is what's in /etc/fstab

```
UUID=32AC-72EA          /boot       vfat        nodev,nosuid,noexec,rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro    0 2
```

The UUID is correct, I've checked it.

journalctl:

```
Sep 24 18:45:31 StoraArch systemd[1]: Unmounting /boot...
Sep 24 18:45:31 StoraArch systemd[1]: Unmounted /boot.
```

**Comment by falconindy** [comment]
I'm guessing you're using EFI? The mount is probably timing out:

https://github.com/systemd/systemd/blob/master/src/gpt-auto-generator/gpt-auto-generator.c#L525

You can verify by looking at `systemctl show -p TimeoutIdleUSec boot.automount`.

You don't need/want this mounted all the time, and it'll be remounted on access. If you really want to disable this behavior, create a dropin at `/etc/systemd/system/boot.automount.d/timeout.conf` with the contents:

```
[Automount]
TimeoutIdleUSec=0
```

**Comment by roberdorfer** [comment]
Ah, that makes sense. I am indeed using EFI. Timeout is set to two minutes. Is there a specific reason to unmount it? Especially since journalctl says it's been unmounted incorrectly and tells me to run fsck on it. Oh, and the automatic mounting on access doesn't work.

```
cd /boot
```

tells me:

```
No such file or directory
```

**Comment by falconindy** [comment]
`cd /boot` fails? And that's the exact error message? Doesn't that mean you somehow have no /boot directory at all?

**Comment by roberdorfer** [comment]
This is the exact output ls -lah while /boot is not mounted.

```
drwxr-xr-x   2 root root    0 Sep 24 14:15 boot/
```

So yes, it's definitely there.

edit: No idea why it failed before, now I can cd into it just fine...

**Comment by roberdorfer** [comment]
I just tried creating the timeout.conf, but the idle time is still set to 2min.

**Comment by mbiebl** [comment]
Seems like other users have the same problem:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=797326

**Comment by kaysievers** [comment]
Why do you expect automounts to be involved? Stuff listen in fstab should not get automatically created automounts.

Do you see something similar to the other bug report:

> systemd[1]: boot.mount: Unit is bound to inactive unit dev-disk-by\x2duuid-....device. Stopping, too.
> systemd[1]: Unmounting /boot...

Then it looks like it would be a question why the device goes away, and not the timeout of the (not active) automount.

**Comment by roberdorfer** [comment]
I guess it was just because of the automount, which I hadn't expected to be a systemd feature. I still don't understand the point in unmounting it, but I guess if this is the intended behavior then this issue can be closed.

**Comment by poettering** [comment]
@RoBi89 basically, the ESP file sytem should be handled very carefully, and we should make sure it is in a clean state if possible. This means we really should unmount it when we don't need it anymore, and hence that's what we do. Or to turn this around: the time window when the ESP is in "dirty" state, i.e. in the porcess of being written to should be short, and hence keeping it mounted during the whole runtime of the system is not ideal, in particular because the ESP something that need to be touched only very seldom, and most of the time is simply not accessed.

**Comment by kaysievers** [comment]
You have an entry in fstab. Do you really still get the automount activated by systemd? That would be a bug, fstab entries take precedence, and no automount should be set up.

**Comment by mbiebl** [comment]
@poettering I think this bug report should be reopened.
That unmount of /boot apparently happens for users which have an fstab entry for /boot.

**Comment by roberdorfer** [comment]
I do have an fstab entry, it looks like this, I hope that helps:

```
# /dev/sda1
UUID=32AC-72EA          /boot       vfat        nodev,nosuid,noexec,rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro    0 2
```

And @poettering thanks for the explanation.

**Comment by heftig** [comment]
I can confirm this; having a fstab entry for /boot does not prevent systemd from starting the automount and unmounting /boot after a timeout.

```
PARTLABEL=strike-boot   /boot   vfat    nosuid,nodev,noexec,noatime,fmask=0133  0       2
```

```
systemd-1 on /boot type autofs (rw,relatime,fd=21,pgrp=1,timeout=120,minproto=5,maxproto=5,direct)
/dev/sda1 on /boot type vfat (rw,nosuid,nodev,noexec,noatime,fmask=0133,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro)
```

However, this could be a feature. IIRC autounmount was introduced because it forces vfat to sync to disk, which tends to avoid corruption.

**Comment by arvidjaar** [comment]
> However, this could be a feature

Ignoring explicit administrator configuration?

The problem is that gpt auto generator never even checks /etc/fstab in the first place. It only checks that /boot is empty or is already mounted, but it is always empty and is never mounted when generators run on boot.

**Comment by heftig** [comment]
Perhaps the fstab generator should output Conflicts on the automount units of mounts that aren't auomounted.

**Comment by kaysievers** [comment]
> The problem is that gpt auto generator never even checks /etc/fstab in the first place.

Generators usually do not check overriding themselves. That is done with priorities of the directories the generators place their stuff. Seem the stuff from fstab generator does not override the gpt generator, or most likely, the automount from gpt generator leaks into the fstab generator one.

**Comment by arvidjaar** [comment]
> Seem the stuff from fstab generator does not override the gpt generator

Of course not. fstab-generator does not emit automount unit, so whatever gpt-generator creates is always used.

I do not see how it can work either. May be gpt-generator and fstab-generator should really be merged in one common generator. After all, their goal is the same - to produce mount units.

**Comment by kaysievers** [comment]
> I do not see how it can work either.

Well, it should work fine for all normal units of the same type. See "man systemd.generator". The fstab units are in a different directory and override the gpt ones.

Seems, we need to find a solution to ignore the automount here which should be disabled as long as with the mount that is disabled by fstab.

At least as long as we split mounts and automounts in separate units, which I don't remember now why we needed to do that.

**Comment by poettering** [comment]
Fixed via #1425. Closing.

**Comment by pyite** [comment]
You need to stop systemd from touching /boot - this is a total disaster.
